File: year/2024/02/solve.py
with open("input.txt", 'r') as f:
    levels = [[*map(int, line.strip().split())] for line in f.readlines()] # O(n)

def check(line):
    if sorted(line) != line and sorted(line, reverse=True) != line: # O(len(line) log(len(line))) = const
        return False
    for i in range(len(line)-1):
        if not 0<abs(line[i]-line[i+1])<4:
            return False
    return True

# *
count = sum(check(line) for line in levels) # O(n)
print(count)

# **
def check2(line):
    if check(line):
        return True

    for i in range(len(line)): # O(len(line)^2 log(len(line))) = const
        new_line = [e for j, e in enumerate(line) if j!=i]
        if check(new_line):
            return True
    return False

count = sum(check2(line) for line in levels) # O(n)
print(count)

File: year/2024/02/solve.jl
input = readlines("input.txt")
lines = input.|>l->parse.(Int, split(l))

# *
check(l) = (issorted(l)||issorted(l, rev=true))&&all(0 .<extrema(abs, diff(l)).<4)
sum(check, lines) |> println

# **
check2(l) = check(l) || any(1:length(l).|>i->check([l[j] for j=1:length(l) if j!=i]))
sum(check2, lines) |> println
